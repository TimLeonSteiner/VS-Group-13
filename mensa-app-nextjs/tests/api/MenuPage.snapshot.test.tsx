import { render, screen } from '@testing-library/react';
import MenuPage from '@/app/menu/page';

// Mock der globalen fetch-Funktion
global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve([
      { _id: 'dish1', name: 'Test-Gericht', price: '€5.00', description: 'Ein leckerer Test.' }
    ]),
  })
) as jest.Mock;

describe('MenuPage Snapshot Test', () => {
  it('sollte nach dem Laden der Daten mit dem Snapshot übereinstimmen', async () => {
    // Vorbereiten
    const { container } = render(<MenuPage />);

    // Ausführen: Warten, bis die Daten geladen sind
    await screen.findByText('Test-Gericht');

    // Überprüfen
    expect(container).toMatchSnapshot();
  });
});