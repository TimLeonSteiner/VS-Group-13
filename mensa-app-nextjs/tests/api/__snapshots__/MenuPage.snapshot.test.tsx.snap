// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuPage Snapshot Test sollte nach dem Laden der Daten mit dem Snapshot übereinstimmen 1`] = `
<div>
  <section
    class="dish-list"
  >
    <article
      class="dish-item"
    >
      <h2>
        Test-Gericht
      </h2>
      <p
        class="price"
      >
        €5.00
      </p>
      <p
        class="description"
      >
        Ein leckerer Test.
      </p>
      <form
        style="margin-top: 1rem;"
      >
        <div>
          <strong>
            Rate this:
          </strong>
          <span
            style="cursor: pointer; font-size: 2rem; color: lightgray;"
          >
            ★
          </span>
          <span
            style="cursor: pointer; font-size: 2rem; color: lightgray;"
          >
            ★
          </span>
          <span
            style="cursor: pointer; font-size: 2rem; color: lightgray;"
          >
            ★
          </span>
          <span
            style="cursor: pointer; font-size: 2rem; color: lightgray;"
          >
            ★
          </span>
          <span
            style="cursor: pointer; font-size: 2rem; color: lightgray;"
          >
            ★
          </span>
        </div>
        <button
          disabled=""
          type="submit"
        >
          Submit Rating
        </button>
      </form>
    </article>
  </section>
</div>
`;
